packages/@ember/-internals/glimmer/lib/component-managers/custom.ts

```diff
@@ -285,6 +285,10 @@ export default class CustomComponentManager<ComponentInstance>
         instance: component,
         template: definition.template,
       });
+
+      registerDestructor(bucket, () => {
+        env.extra.debugRenderTree.willDestroy(bucket);
+      });
```

Comment:
Maybe I'm missing something here, we used to only do the work to track / remove stuff when `ENV._DEBUG_RENDER_DEBUG` was set. Has that changed?

Reply:
Same as above, there was always some setup code in the `create` hook so we can combine it with that